### PR TITLE
Add nameFilter field for the AlertList panel.

### PR DESCRIPTION
**Pull request by dafna-starkware**
## What does this do?
Add nameFilter field for the AlertList panel.

## Why is it a good idea?
Enable to filter alerts by their name in an AlertList panel.

---
This change is [Reviewable](https://reviewable.io/reviews/weaveworks/grafanalib/338)

### CODE REVIEW

**Review by JamesGibo — APPROVED** [review]
Thanks for this PR and looks good to me.
I wondered if while making this change, if you want to add some documentation to the AlertList class and or some simple unit tests for it?

**Review by OmriEshhar1 — COMMENTED** [review]
> Reviewed 2 of 2 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/weaveworks/grafanalib/338)* status: all files reviewed, 9 unresolved discussions (waiting on @dafna-starkware)

___
*[grafanalib/core.py, line 1474 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX2-uyq2aGeLXO5RFmu:-MX2-uyq2aGeLXO5RFmv:b-nyd31h) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1474)):*
> ```Python
> class AlertList(object):
>     """Generates the AlertList Panel.
> ```

some parameters are not documented. is it intentional?
___
*[grafanalib/core.py, line 1477 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX20BmgEn-sHdfJPdym:-MX20BmgEn-sHdfJPdyn:bjk3hg5) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1477)):*
> ```
> If to show only alerts from the current dashboard.
> ```
change to something like:
If true, shows only alerts from...
___
*[grafanalib/core.py, line 1480 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX20YlB8EGrrYaUefMB:-MX20YlB8EGrrYaUefMC:b2d2na) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1480)):*
> ```
> According to which order to sort the alerts:
> ```
change to something like:
Defines the sorting order of the alerts: 
___
*[grafanalib/core.py, line 1483 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX20u0wCUI4TLmNFmRY:-MX20u0wCUI4TLmNFmRZ:bdx263c) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1483)):*
> ```
> A list of the statuses of the alerts to display in the list.
> ```
change to something like:
Show alerts with statuses from the stateFilter list.
___
*[grafanalib/core.py, line 1487 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX222rQCaFODL5JMiLV:-MX222rQCaFODL5JMiLW:b-96pequ) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1487)):*
> ```
> An empty list means all the statuses.
> ```
An empty list means show all alerts.
___
*[grafanalib/core.py, line 1488 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX22BnX7_NRB-qtlXAs:-MX22BnX7_NRB-qtlXAt:b-7ylpii) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1488)):*
> ```Python
>          ALERTLIST_STATE_PAUSED, ALERTLIST_STATE_EXECUTION_ERROR, ALERTLIST_STATE_PENDING].
>         An empty list means all the statuses.
>     :param title: panel title.
> ```

panel -> Panel
___
*[grafanalib/core.py, line 1489 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX22Gvn6La5kBPBGxLy:-MX22Gvn6La5kBPBGxLz:b-xz5wwg) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1489)):*
> ```
> Defines if to display the panel without a background.
> ```
change to something like:
If true, display the panel without a background.
___
*[grafanalib/core.py, line 1490 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX22TbM4WPlWdXg38X0:-MX22TbN64lxAORFpguP:bww0fyj) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1490)):*
> ```
> Show only alerts that contain this string in their name.
> ```
Show only alerts that contain nameFilter in their name.
___
*[grafanalib/core.py, line 1504 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX22eSTD7thXRw2eOYq:-MX22eSTD7thXRw2eOYr:bnqynki) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1504)):*
> ```Python
>     title = attr.ib(default="")
>     transparent = attr.ib(default=False, validator=instance_of(bool))
>     nameFilter = attr.ib(default="", validator=instance_of(str))
> ```

while you're at it, you can add additional properties that are missing for AlertList panel, e.g. dashboardFilter, dashboardTags, ...

**Review by dafna-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/weaveworks/grafanalib/338)* status: 1 of 2 files reviewed, 8 unresolved discussions (waiting on @OmriEshhar1)

___
*[grafanalib/core.py, line 1474 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX2-uyq2aGeLXO5RFmu:-MX2CjTN0jy-TD6pLKhS:b-896fix) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1474)):*
<details><summary><i>Previously, OmriEshhar1 wrote…</i></summary><blockquote>

some parameters are not documented. is it intentional?
</blockquote></details>

Done.
___
*[grafanalib/core.py, line 1477 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX20BmgEn-sHdfJPdym:-MX2CtTVCn2psWWoUTrg:b-896fix) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1477)):*
<details><summary><i>Previously, OmriEshhar1 wrote…</i></summary><blockquote>

> ```
> If to show only alerts from the current dashboard.
> ```
change to something like:
If true, shows only alerts from...
</blockquote></details>

Done.
___
*[grafanalib/core.py, line 1480 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX20YlB8EGrrYaUefMB:-MX2DFOM1XCSUhfLw5QZ:b-896fix) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1480)):*
<details><summary><i>Previously, OmriEshhar1 wrote…</i></summary><blockquote>

> ```
> According to which order to sort the alerts:
> ```
change to something like:
Defines the sorting order of the alerts: 
</blockquote></details>

Done.
___
*[grafanalib/core.py, line 1483 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX20u0wCUI4TLmNFmRY:-MX2Du1DAqEs06aXuJpO:b-896fix) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1483)):*
<details><summary><i>Previously, OmriEshhar1 wrote…</i></summary><blockquote>

> ```
> A list of the statuses of the alerts to display in the list.
> ```
change to something like:
Show alerts with statuses from the stateFilter list.
</blockquote></details>

Done.
___
*[grafanalib/core.py, line 1487 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX222rQCaFODL5JMiLV:-MX2E0MABhEdWRiF6XpB:b-896fix) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1487)):*
<details><summary><i>Previously, OmriEshhar1 wrote…</i></summary><blockquote>

> ```
> An empty list means all the statuses.
> ```
An empty list means show all alerts.
</blockquote></details>

Done.
___
*[grafanalib/core.py, line 1488 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX22BnX7_NRB-qtlXAs:-MX2E7Zd0sOohIzDpw-y:b-896fix) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1488)):*
<details><summary><i>Previously, OmriEshhar1 wrote…</i></summary><blockquote>

panel -> Panel
</blockquote></details>

Done.
___
*[grafanalib/core.py, line 1489 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX22Gvn6La5kBPBGxLy:-MX2EDZdCdTYDkOEMEPK:b-896fix) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1489)):*
<details><summary><i>Previously, OmriEshhar1 wrote…</i></summary><blockquote>

> ```
> Defines if to display the panel without a background.
> ```
change to something like:
If true, display the panel without a background.
</blockquote></details>

Done.
___
*[grafanalib/core.py, line 1490 at r1](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MX22TbM4WPlWdXg38X0:-MX2EY1X6MjElUf_uzT_:b-896fix) ([raw file](https://github.com/weaveworks/grafanalib/blob/b5adb29ff25af2157eac91b522f4a87e30564783/grafanalib/core.py#L1490)):*
<details><summary><i>Previously, OmriEshhar1 wrote…</i></summary><blockquote>

> ```
> Show only alerts that contain this string in their name.
> ```
Show only alerts that contain nameFilter in their name.
</blockquote></details>

Done.

**Review by OmriEshhar1 — COMMENTED** [review]
:lgtm:

> Reviewed 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/weaveworks/grafanalib/338#-:-MX2PiKz4IQiDyl3kQak:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @dafna-starkware)

**Review by OmriEshhar1 — COMMENTED** [review]
> Reviewed 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/weaveworks/grafanalib/338)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @dafna-starkware)

File: grafanalib/core.py [hunk 1]
```diff
@@ -1469,12 +1470,40 @@ def to_json_data(self):
...
+        default=attr.Factory(list),
+        validator=attr.validators.deep_iterable(attr.validators.instance_of(str)))
     description = attr.ib(default="")
+    gridPos = attr.ib(default=None)
     id = attr.ib(default=None)
     limit = attr.ib(default=DEFAULT_LIMIT)
-    links = attr.ib(default=attr.Factory(list))
+    links = attr.ib(
+        default=attr.Factory(list),
+        validator=attr.validators.deep_iterable(attr.validators.instance_of(DataLink)))
```
Comment (JamesGibo) [inline grafanalib/core.py:1506]:
```suggestion
        validator=attr.validators.deep_iterable(
            attr.validators.instance_of(DataLink),
            attr.validators.instance_of(list)
       )
   )
```
https://www.attrs.org/en/stable/api.html#attr.validators.deep_iterable

**Review by OmriEshhar1 — COMMENTED** [review]
> Reviewed 1 of 1 files at r6, 1 of 1 files at r7.
> *[Reviewable](https://reviewable.io/reviews/weaveworks/grafanalib/338)* status: all files reviewed, 1 unresolved discussion (waiting on @dafna-starkware)

**Review by dafna-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/weaveworks/grafanalib/338)* status: all files reviewed, 1 unresolved discussion (waiting on @JamesGibo)

___
*[grafanalib/core.py, line 1506 at r6](https://reviewable.io/reviews/weaveworks/grafanalib/338#-MWzWElManELIG-jttQ5-r6-1506:-MX60nV33jO_eYBeAy0h:b-896fix) ([raw file](https://github.com/weaveworks/grafanalib/blob/b682a1520d70bcdc36c6593ab3f988e95780c68a/grafanalib/core.py#L1506)):*
<details><summary><i>Previously, JamesGibo wrote…</i></summary><blockquote>

```suggestion
        validator=attr.validators.deep_iterable(
            attr.validators.instance_of(DataLink),
            attr.validators.instance_of(list)
        )
    )
```
https://www.attrs.org/en/stable/api.html#attr.validators.deep_iterable
</blockquote></details>

Done.

**Pull request merged**